### PR TITLE
add new Swiftlint Rules

### PR DESCRIPTION
**Pull request by konDeichmann**
Please consider using the following Rules in the Frameworks only,
They seem not to work well in the Apps

```
missing_docs
explicit_enum_raw_value
*_contents_order
```

### CODE REVIEW

File: SwiftLint/swiftlint.yml [hunk 1]
```diff
@@ -54,6 +55,39 @@ opt_in_rules:
...
+  - redundant_type_annotation
+  - sorted_first_last
+  - static_operator
+  - toggle_bool
+  - unowned_variable_capture
+  - unused_import
+  - vertical_parameter_alignment_on_call
+  - vertical_whitespace_closing_braces
+  - convenience_type
+  - number_separator
```
Comment (Tomusm) [inline SwiftLint/swiftlint.yml]: This creates a lot of warnings on HiDrive. Sometimes weird situations, for example for a color declaration
- `UIColor(red: 0.9294117647, green: 0.9294117647, blue: 0.9294117647, alpha: 1.0)` is triggered for example.
Overall, nice to have but maybe for later ?!

File: SwiftLint/swiftlint.yml [hunk 2]
```diff
@@ -54,6 +55,39 @@ opt_in_rules:
...
+  - multiline_parameters_brackets
+  - object_literal
+  - overridden_super_call
+  - redundant_type_annotation
+  - sorted_first_last
+  - static_operator
+  - toggle_bool
+  - unowned_variable_capture
+  - unused_import
+  - vertical_parameter_alignment_on_call
```
Comment (Tomusm) [inline SwiftLint/swiftlint.yml]: On HiDrive iOS, this is triggered although the parameters are aligned

File: SwiftLint/swiftlint.yml [hunk 3]
```diff
@@ -54,6 +55,39 @@ opt_in_rules:
...
+  - joined_default_parameter
+  - last_where
+  - legacy_multiple
+  - literal_expression_end_indentation
+  - lower_acl_than_parent
+  - modifier_order
+  - multiline_arguments
+  - multiline_function_chains
+  - multiline_literal_brackets
+  - multiline_parameters_brackets
```
Comment (Tomusm) [inline SwiftLint/swiftlint.yml]: This one is triggered for:
```		
func cell(_ backupObject: BackupObject?,
				  viewController: HDBRRestoreViewController,
				  tableView: UITableView,
				  atIndexPath: IndexPath,
				  restoreProgressBlock: ((_ progress: BackupProgress?) -> Void)? = nil,
				  restoreFinishedBlock: ((_ report: BackupProcessReport?, _ error: BusinessError?) -> Void)? = nil) -> UITableViewCell {
```

To fix it, we would have to write:

```
		func cell(
			_ backupObject: BackupObject?,
				  viewController: HDBRRestoreViewController,
				  tableView: UITableView,
				  atIndexPath: IndexPath,
				  restoreProgressBlock: ((_ progress: BackupProgress?) -> Void)? = nil,
				  restoreFinishedBlock: ((_ report: BackupProcessReport?, _ error: BusinessError?) -> Void)? = nil
			) -> UITableViewCell {
```

Which looks horrible to me

File: SwiftLint/swiftlint.yml [hunk 4]
```diff
@@ -54,6 +55,35 @@ opt_in_rules:
...
+  - joined_default_parameter
+  - last_where
+  - legacy_multiple
+  - literal_expression_end_indentation
+  - lower_acl_than_parent
+  - modifier_order
+  - multiline_arguments
+  - multiline_function_chains
+  - multiline_literal_brackets
+  - object_literal
```
Comment (konDeichmann) [inline SwiftLint/swiftlint.yml]: remove this one
```suggestion
  - convenience_type
```

**Pull request merged**